ivadomed/training.py

```diff
@@ -198,7 +198,9 @@ def train(model_params, dataset_train, dataset_val, training_params, log_directo
                 # METRICS COMPUTATION
                 gt_npy = gt_samples.cpu().numpy().astype(np.uint8)
                 preds_npy = preds.data.cpu().numpy()
-                metric_mgr(preds_npy.astype(np.uint8), gt_npy)
+
+                if dataset_train.task == "segmentation":
+                    metric_mgr(preds_npy.astype(np.uint8), gt_npy)
```

Comment:
I would suggest to make the change here instead: [here](https://github.com/neuropoly/ivado-medical-imaging/blob/master/ivadomed/main.py#L72). ie define metrics of interest for a given task.